10_rest/n0npax/pkg/puppy/rest.go

```diff
@@ -0,0 +1,90 @@
+package puppy
+
+import (
+	"strconv"
+
+	"github.com/gin-gonic/gin"
```

Comment:
we have been using `chi` a lot internally - but why not try something else?
is there a specific reason for choosing gin?

Reply:
It's one of the most popular frameworks. When I was googling I found `gin` and `revel` are most common frameworks